doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,530 @@
+    expressions are not supported.
+
+
+Simulation of mid-circuit measurements
+--------------------------------------
+
+PennyLane currently offers three methods to simulate mid-circuit measurements
+on classical computers: the deferred measurements principle, dynamic one-shot
+sampling, and a tree-traversal approach. These methods differ in their memory requirements
+and computational cost, as well as their compatibility with other features such as
+shots and differentiation methods.
+While the requirements depend on details of the simulation, the expected 
+scalings are
+
+.. role:: gr
+.. role:: or
+.. role:: rd
+
+.. raw:: html
+
+   <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
+   <script>
+     $(document).ready(function() {
+       $('.gr').parent().parent().addClass('gr-parent');
+       $('.or').parent().parent().addClass('or-parent');
+       $('.rd').parent().parent().addClass('rd-parent');
+     });
+   </script>
+   <style>
+       .gr-parent {background-color:#e1eba8}
+       .or-parent {background-color:#ffe096}
+       .rd-parent {background-color:#ffb3b3}
+       .tb { border-collapse: collapse; }
+       .tb th, .tb td { padding: 1px; border: solid 1px black; vertical-align: middle; column-width: auto; }
+   </style>
+
+.. rst-class:: tb
+
++--------------------------+-------------------------------------------+------------------------------------------------+-------------------------------------------+----------------------------------+
+| **Simulation technique** | **Memory**                                | **Time**                                       | **Differentiation support**               | **Supports shots/analytic mode** |
++==========================+===========================================+================================================+===========================================+==================================+
+| Deferred measurements    | :rd:`\ ` :math:`\mathcal{O}(2^{n_{MCM}})` | :rd:`\ ` :math:`\mathcal{O}(2^{n_{MCM}})`      | :gr:`\ ` yes                              | :gr:`\ ` yes / yes               |
++--------------------------+-------------------------------------------+------------------------------------------------+-------------------------------------------+----------------------------------+
+| Dynamic one-shot         | :gr:`\ ` :math:`\mathcal{O}(1)`           | :rd:`\ ` :math:`\mathcal{O}(n_{shots})`        | :or:`\ ` finite differences\ :math:`{}^1` | :or:`\ ` yes / no                |
++--------------------------+-------------------------------------------+------------------------------------------------+-------------------------------------------+----------------------------------+
+| Tree-traversal           | :or:`\ ` :math:`\mathcal{O}(n_{MCM}+1)`   | :or:`\ ` :math:`\mathcal{O}(\leq 2^{n_{MCM}})` | :or:`\ ` finite differences\ :math:`{}^1` | :or:`\ ` yes / no                |
```

Comment:
It can also be less than both. For instance, running iterative QPE with 26 iterations (mcms) and 100k shots, it explored ~26k branches instead of 100k (for shots) or 67108864 (for 2**mcms).

Reply:
>Using "O" implies complexity to be asymptotically less than or equal to the inside function anyway.

Yeah, I was trying to reflect that the scaling is advantageous, so I wanted it to differ from deferred measurements :D With @vincentmr 's suggestion, this is accomplished :) 

> $\mathcal{O}(min(n_{shots}, 2^{n_{MCM}}))$

I like it!

Reply:
Thanks :)